kvstore/kvstore_server.cpp

```diff
@@ -1,18 +1,30 @@
-#include <kvstore_server.h>
+#include kvstore_server.h
 
 Status KvstoreServiceImpl::put(ServerContext* context, const PutRequest* request, PutReply* reply) {
-  // TODO: put
+  std::unique_lock<std::mutex> lock(mu_);
```

Comment:
does this compile? looks like you moved your mutex to the kvmap so you should lock the mutex at the top of get, put, remove in kvmap not here.

Reply:
Moved and compiles